src/ai/backend/manager/server.py

```diff
@@ -281,6 +292,83 @@ async def manager_status_ctx(root_ctx: RootContext) -> AsyncIterator[None]:
     yield
 
 
+@actxmgr
+async def leader_election_ctx(root_ctx: RootContext) -> AsyncIterator[None]:
+    configuration = root_ctx.local_config["manager"]["raft-configuration"]
+    raft_id = configuration[root_ctx.pidx]
+    grpc_port = int(raft_id.split(":")[-1])
```

Comment:
Thanks for introducing `str.partition()`!
However, I found that it would be better to use `HostPortPair` type, which has a `port` field, for "raft-configuration".  Maybe I can have a chance to use `str.partition()` in other tasks. 😢

About the raft component and manager, it is using a callback for getting notified of its raft status change.
Event handlers are used between a manager server and its `GlobalTimer` objects.
```
Raft ---(callback)---> Manager <---(event)---> GlobalTimer(a)
                                      ㄴ-----> GlobalTimer(b)
```